test/e2e/gateway/gatewayapi/gateway_api.go

```diff
@@ -115,6 +116,67 @@ func GatewayAPI() {
 		return ip
 	}
 
+	Describe("GatewayClass parametersRef", Ordered, func() {
+		haGatewayClass := `
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: GatewayClass
+metadata:
+  name: kuma
+spec:
+  controllerName: gateways.kuma.io/controller
+  parametersRef:
+    kind: MeshGatewayConfig
+    group: kuma.io
+    name: ha-config`
+
+		haConfig := `
+apiVersion: kuma.io/v1alpha1
+kind: MeshGatewayConfig
+metadata:
+  name: ha-config
+spec:
+  replicas: 3`
+
+		haGateway := `
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: Gateway
+metadata:
+  name: kuma
+  namespace: kuma-test
+spec:
+  gatewayClassName: kuma
+  listeners:
+  - name: proxy
+    port: 8080
+    protocol: HTTP`
+
+		BeforeAll(func() {
+			E2EDeferCleanup(func() error {
+				return k8s.RunKubectlE(cluster.GetTesting(), cluster.GetKubectlOptions(TestNamespace), "delete", "gateway", "kuma")
+			})
+			Expect(YamlK8s(haConfig)(cluster)).To(Succeed())
+			// TODO this shouldn't be necessary, need to reconcile on MeshGatewayConfig
+			time.Sleep(10 * time.Second)
```

Comment:
How can we fix it?

Reply:
Ah I think this can be removed with the last commit actually